L-stick or Light stick

Frost-wood opened the pull request:
Basically a flashlight on a stick-but made for combat. Has long lasting batteries, but I balanced it out by making it rare and uncraftable.  Has same combat stats as a quarterstaff, but lighter due to the superalloy materials.

Made by Light corporation, whose products are expensive(think apple), but quality.

Rivet-the-Zombie commented:
Shirak!

mugling commented:
Would this not largely duplicate the intention of the heavy duty flashlight?

Coolthulhu commented:
* Weight reduction can lead to significantly lower move cost per attack, you should balance `damage / move_cost` rather than pure damage.
* That 4 to-hit is huge. 3 is totally enough.
* That battery efficiency is way, way, WAY too huge. 120 turns of LIGHT_500 per charge? A regular flashlight has 20 turns of LIGHT_300, heavy-duty has 12 turns of LIGHT_500.
* Weight 5? 5 GRAMS? That's just 2.5 tablespoons of aluminum. And superalloys are titanium, which is heavier than aluminum.

Coolthulhu commented on data/json/items/tools.json:
```diff
@@ -11073,5 +11073,68 @@
+    "bashing": 20,
+    "cutting": 0,
+    "to_hit": 4,
+	"category" : "weapons",
```
Tabs.
Should be 4 spaces
There are few more places where you used tabs, check all the files (if you're using notepad++, you can search for tabs in regular CTRL+F).

Frost-wood replied:
thought I got all of them.

Frost-wood commented:
@Coolthulhu -the weight doesn't seem to go below 2lbs no matter what I set the weight at... bug?  I changed the weight to 1000, and it weighs 4.41lbs.
-changed the battery time to 40 turns.

Coolthulhu commented:
Could be a display bug, try checking in metric weights.

Coolthulhu commented on data/json/items/tools.json:
```diff
@@ -11073,5 +11073,68 @@
+    "description": "Produced by the Light corporation, this stylish staff not only functions as a light source, but doubles as a light-weight weapon, due to the superalloy materials it is made of.  The staff uses long-lasting fuel cells as its power source.  The staff is glowing brightly, which is slowly draining the batteries",
+    "price": 5000,
+    "material": ["superalloy", "steel"],
+	"techniques" : ["WBLOCK_1", "RAPID"],
```
Tabs

Coolthulhu commented on data/json/items/tools.json:
```diff
@@ -11073,5 +11073,68 @@
+    "to_hit": 4,
+    "category" : "weapons",
+    "max_charges": 1000,
+    "initial_charges": 1000,
```
Those initial charges could be lower, it's already crazy good without free 1k batteries.

Coolthulhu commented on data/json/items/tools.json:
```diff
@@ -11073,5 +11073,68 @@
+    "volume": 12,
+    "bashing": 20,
+    "cutting": 0,
+    "to_hit": 4,
```
I don't see any justification for it having 4 to-hit.

Coolthulhu commented:
OK, maybe not crazy good, but it's still a really solid drop that isn't supposed to be a dedicated weapon, is in semi-common groups, is much better than most bashing weapons (due to strength cap) and obsoletes two classes of items (flashlights and bashing-only weapons) when it drops.

Frost-wood commented:
lowered the initial charges to 0, to hit to 3, lowered bash damage to be slightly lower than the quarterstaff.

Coolthulhu commented:
Now it looks good mechanically.

The description should mention that the flashlight part is efficient. "Long lasting batteries" doesn't work well here since it can be reloaded with any batteries.

kevingranade commented:
The high weight might be because of the batteries, I can't recall if they
get added in like that or not, but it would make sense.

Frost-wood commented:
You're right I didn't think about that.  The attack speed actually decreases as you add batteries.

mugling commented on data/json/items/tools.json:
```diff
@@ -11073,5 +11073,68 @@
+    "ammo": "battery",
+    "revert_to": "l-stick",
+    "use_action": {
+        "type": "auto_transform",
```
You can use `transform` instead of `autotransform` here, see #14152
